File: app/src/main/java/com/silentgames/silent_planet/mvp/main/MainContract.java
package com.silentgames.silent_planet.mvp.main;

import android.app.Activity;
import android.content.res.Resources;
import android.graphics.Bitmap;

import com.silentgames.silent_planet.model.Cell;
import com.silentgames.silent_planet.model.cells.CellType;
import com.silentgames.silent_planet.model.entities.EntityType;
import com.silentgames.silent_planet.mvp.BasePresenter;

import java.util.List;

public interface MainContract {
    interface View {

        void drawGrid();

        void drawBattleGround(Cell[][] gameMatrix);

        void reDraw(int eventX,int eventY, Bitmap entity);

        void showToast(String text);

        void showObjectIcon(CellType cellType);

        void showObjectIcon(EntityType entityType);

        void showCellListItem(int x, int y, List<String> playerList);

        void hideCellListItem();

        void update(Runnable runnable);

        void enableButton(boolean isEnabled);

        void setImageCrystalText(String text);

    }
    interface Presenter extends BasePresenter {

        void onSingleTapConfirmed(int x, int y);

        void onCellListItemSelectedClick(int x, int y, String text);

        void onActionButtonClick();

    }
}

File: app/src/main/java/com/silentgames/silent_planet/mvp/main/MainContract.kt
package com.silentgames.silent_planet.mvp.main

import android.graphics.Bitmap
import com.silentgames.silent_planet.dialog.EntityData
import com.silentgames.silent_planet.logic.ecs.Axis
import com.silentgames.silent_planet.logic.ecs.component.FractionsType
import com.silentgames.silent_planet.mvp.BasePresenter

interface MainContract {
    interface View {

        fun showToast(text: String)

        fun fillEntityName(text: String)

        fun fillDescription(text: String)

        fun showObjectIcon(bitmap: Bitmap)

        fun enableButton(isEnabled: Boolean)

        fun setImageCrystalText(text: String)

        fun showEntityMenuDialog(entityList: MutableList<EntityData>, currentCell: EntityData)

        fun changeAlienCristalCount(crystals: Int)

        fun changeHumanCristalCount(crystals: Int)

        fun changePirateCristalCount(crystals: Int)

        fun changeRobotCristalCount(crystals: Int)

        fun selectCurrentFraction(fractionType: FractionsType)

        fun showPlayerBuybackSuccessMessage(name: String)

        fun showPlayerBuybackFailureMessage(missingAmount: Int)

    }

    interface Presenter : BasePresenter {

        fun onSingleTapConfirmed(axis: Axis)

        fun onActionButtonClick()

        fun onEntityDialogElementSelect(entityData: EntityData)

        fun onCapturedPlayerClick(entityData: EntityData)

    }
}
